atomic_reactor/__init__.py

```diff
@@ -32,6 +35,28 @@ def format(self, record):
         return super(ArchFormatter, self).format(record)
 
 
+class EncodedStream(object):
+    def __init__(self, fileno, encoding):
+        self.binarystream = fdopen(dup(fileno), 'wb')
+        self.encoding = encoding
+
+    def write(self, text):
+        if not isinstance(text, bytes):
+            self.binarystream.write(text.encode(self.encoding))
+        else:
+            self.binarystream.write(text)
+
+    def __del__(self):
+        try:
+            self.binarystream.close()
+        except AttributeError:
+            pass
+
+
+def get_encoding(name=None):
```

Comment:
Wonder if it might be nice to use the same default value set_logging() uses here.